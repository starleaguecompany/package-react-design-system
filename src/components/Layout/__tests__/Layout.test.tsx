import * as React from 'react'
import { render, screen } from '@testing-library/react'

import { Layout } from '..'

describe('Layout', () => {
  test('should render correctly without prop', () => {
    render(<Layout />)

    expect(screen.getByTestId('Layout')).toMatchSnapshot()
    expect(screen.getByTestId('Layout')).toHaveClass('container', { exact: true })
  })

  test('should render correctly with attributes', () => {
    render(<Layout id="test-id" className="test-class" />)

    expect(screen.getByTestId('Layout')).toMatchSnapshot()
    expect(screen.getByTestId('Layout')).toHaveClass('test-class')
    expect(screen.getByTestId('Layout')).toHaveAttribute('id', 'test-id')
  })
})
