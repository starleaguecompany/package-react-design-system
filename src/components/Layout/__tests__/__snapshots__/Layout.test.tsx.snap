// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render correctly with attributes 1`] = `
<div
  class="test-class container"
  data-qa="Layout"
  id="test-id"
/>
`;

exports[`Layout should render correctly without prop 1`] = `
<div
  class="container"
  data-qa="Layout"
/>
`;
